cfme/middleware/__init__.py

```diff
@@ -1,8 +1,54 @@
 from cfme.web_ui import CheckboxTable, toolbar as tb
 from functools import partial
 
+from cfme.web_ui import Table
+
 cfg_btn = partial(tb.select, 'Configuration')
 mon_btn = partial(tb.select, 'Monitoring')
 pol_btn = partial(tb.select, 'Policy')
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
+
+
+class Validatable(object):
+    """
+    Class which Middleware provider and other middleware pages must extend
+    to be able to validate properties values shown in summary page.
+
+    """
+
+    """
+    Tuples which first value is the provider class's attribute name,
+    the second value is provider's UI summary page field key.
+
+    Should have values in child classes.
+
+    """
+    property_tuples = []
+
+    def properties_ui(self):
+        property_ui = {}
+        for row in Table('//*[@id="main-content"]//div[1]/table').rows():
+            property_ui.update({row[0].text: row[1].text})
+        return property_ui
+
+    def validate_properties(self):
+        """
+        Validation method which checks whether class attributes,
+        which were used during creation of provider,
+        is correctly displayed in Properties section of provider UI.
+        The maps between class attribute and UI property is done via 'property_tuples' variable.
+
+        Returns: pair of validation result as Boolean and failure message
+        """
+        properties = self.properties_ui()
+        if properties:
+            for property_tuple in self.property_tuples:
+                expected_value = str(getattr(self, property_tuple[0]))
+                shown_value = str(properties[property_tuple[1]])
+                if expected_value != shown_value:
+                    return False, 'Property {} has wrong value, expected {} but was {}'.format(
+                        property_tuple[1], expected_value, shown_value)
+            return True, None
+        else:
+            return False, 'No property was found in UI'
```

Comment:
I don't think that the method has to return the tuple, which is highly impractical. I believe according to the python style and our approach something that validates either passes or raises an exception.